File: 3379. Transformed Array/Solution.java
class Solution {
    public int[] constructTransformedArray(int[] nums) {
        int length = nums.length;
        int[] result = new int[length];
        for(int i=0;i<length;i++){
            if(nums[i]==0){
                result[i] = nums[i]; 
            }else if (nums[i] > 0){
                result[i] = nums[(i+nums[i])%length]; 
            }else{
                result[i] = nums[(length + i - (Math.abs(nums[i]) % length))%length];
            }
        }
        return result; 
    }
}

File: 3379. Transformed Array/Solution.py
class Solution:
    def constructTransformedArray(self, nums: List[int]) -> List[int]:
        result = nums.copy()
        n = len(nums)
        for i, x in enumerate(nums): 
            if x == 0: 
                result[i] = x
            elif x > 0: 
                idx = (i + x) % n
                result[i] = nums[idx]
            else: 
                idx = (i - (abs(x) % n) + n) % n
                result[i] = nums[idx]
        return result
